crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -637,6 +659,20 @@ fn write_bridge_config_to_file(
     common_mosquitto_config.serialize(&mut common_draft)?;
     common_draft.persist()?;
 
+    Ok(())
+}
+
+fn write_bridge_config_to_file(
+    config_location: &TEdgeConfigLocation,
+    bridge_config: &BridgeConfig,
+) -> Result<(), ConnectError> {
+    let dir_path = config_location
+        .tedge_config_root_path
+        .join(TEDGE_BRIDGE_CONF_DIR_PATH);
+
+    // This will forcefully create directory structure if it doesn't exist, we should find better way to do it, maybe config should deal with it?
+    create_directories(dir_path)?;
```

Comment:
Yeah. The directories should be created by `tedge init` and `tedge connect` should fail if given a path with unknown parents.